compiler/src/main/java/org/qbicc/graph/Deref.java

```diff
@@ -36,10 +53,14 @@ public boolean equals(final Deref other) {
     }
 
     public ValueType getType() {
-        return value.getType();
+        return toType;
+    }
+
+    public Load getLoad() {
+        return load;
     }
 
     public <T, R> R accept(final ValueVisitor<T, R> visitor, final T param) {
-        return visitor.visit(param, this);
+        return visitor.visit(param, this.getLoad());
```

Comment:
This is not allowed. Either `Deref` is a node of its own, or it's not; by bypassing its visitor method, things which analyze the node graph will likely get confused.